actioncable/app/assets/javascripts/action_cable/events.coffee

```diff
@@ -0,0 +1,6 @@
+class ActionCable.Events extends Event
+
+class ActionCable.Events.Disconnected extends ActionCable.Events
+  constructor: (@code, @reason, @isRecoverable) ->
+
+  @ProtocolNotSupported = new Disconnected(4001, "The client and server could not agree on a subprotocol to use.", false)
```

Comment:
I'm -1 on using a custom Event object to communicate the disconnect info and would prefer a plain object instead.